io/include/detray/io/csv/track_parameters.hpp

```diff
@@ -131,7 +131,8 @@ inline void write_free_track_params(
         for (const auto &track_param : track_params) {
 
             const auto &glob_pos = track_param.pos();
-            const auto &p = track_param.mom();
+            const scalar charge = track_param.qop() < 0.f ? -1.f : 1.f;
```

Comment:
Won't we have to pass the particle hypothesis here, too?

Reply:
track parameter IO is only used for validation purpose with a single charge, so it is not very necessary

Reply:
That is the only current use case, true, but the csv io tool was not necessarily meant to be restricted to that. It could also be used for different work in the future, in which case would  ```q=1``` not become a bug thats hard to find? Lets put at least an assert there.

Since the track generators can now be set to different charges, could you then also put an assert into the detector scan config, so that we are sure the tracks there are always generated with ```q=+-1``` and no inconsistencies appear between the track states in the whiteboard and the csv files?

Reply:
I don't think testing the charges other than +-1 in _helix_ test is going to be useful (because there is no difference from changing the bfield). But I can put an assert

Reply:
BTW can you explain your request clearly? I am not sure writing assert(charge == 1 || charge == -1) makes any sense because I don't know the charge of track parameters.

May I suggest you to define a charge in the base test suites and distribute the charge to its children later?

Reply:
I suggest putting something like:
```
assert(math::fabs(m_cfg.track_generator().charge()) == 1.f * unit<scalar_t>::e && "Detector scan can only be run with particle charge +-1e");
```
somewhere in the beginning of the function ```fill_scan_data()``` in ```detector _scan.hpp```